agent/util-scripts/test-bin/common-tm-cleanup

```diff
@@ -10,12 +10,26 @@ function _kill_pid {
     fi
 }
 
+function _destroy_redis {
+    redis_pids=$(ps uxww | grep -E "[r]edis-server" | awk '{print $2}')
```

Comment:
What is `"[r]edis-server"`?? This is matching a required "set" consisting of exclusively "r". If there's some subtle RE distinction from just "redis-server", I'm missing it.

Reply:
Prevents matching the `grep` process itself.

Reply:
Fascinating. I can see that it does. I can't see why it would. Technically, in strict RE definition, "r" is exactly the same as `[r]`... an RE that matches exactly the single character "r". There's clearly some semantic distinction I'm missing. ☹️

Reply:
The `[r]edis-server` != `redis-server`.  The regex will match `redis-server` but not the text string `[r]edis-server`.

Reply:
This clever little trick is one I remember from the great @agriffis.

Reply:
Ha; of course. Tunnel vision. 🙄